arbos/tx_processor.go

```diff
@@ -514,14 +514,30 @@ func (p *TxProcessor) EndTxHook(gasLeft uint64, success bool) {
 	}
 
 	purpose := "feeCollection"
-	util.MintBalance(&networkFeeAccount, computeCost, p.evm, scenario, purpose)
+	if p.state.FormatVersion() > 4 {
+		infraFeeAccount, err := p.state.InfraFeeAccount()
+		p.state.Restrict(err)
+		if infraFeeAccount != (common.Address{}) {
+			infraFee, err := p.state.L2PricingState().MinBaseFeeWei()
+			p.state.Restrict(err)
+			if arbmath.BigLessThan(gasPrice, infraFee) {
+				infraFee = gasPrice
+			}
+			infraComputeCost := arbmath.BigSub(arbmath.BigMul(infraFee, arbmath.UintToBig(gasUsed)), p.PosterFee)
```

Comment:
This seems incorrect. We're looking at `minGasPrice * gasUsed - posterFee`, but the poster fee is measured in the actual gas price not the minimum gas price. We could do `minGasPrice * (gasUsed - p.posterGas)`. We should also add a check for if this is negative for safety, and some tests would be helpful too.